### ISSUE TITLE
Beta-community-app: When certain combinations of filters are applied, screen flickers continuously

### ISSUE THREAD

**Issue by nithyaasworld** [topcoder-platform/community-app#2392]
**These were the steps I followed before the screen flickering occurs.
But this happens for certain other combination of filters too (which I am not able to reproduce easily)**

1. Go to https://beta-community-app.topcoder.com/challenges
2. Choose Android in the Keyword field
3. Once the results are loaded, remove the Android Keyword
4. Enter "QA" keyword. Once the results are loaded, remove it
5. Then Enter "Bug Hunt" in Subtrack and observe that the screen flickers continuously

**Video**
[2019-05-31_23-42-31.zip](https://github.com/topcoder-platform/community-app/files/3242738/2019-05-31_23-42-31.zip)

**Comment by ThomasKranitsas** [comment]
Contest https://www.topcoder.com/challenges/30092403 has been created for this ticket.<br/><br/>```This is an automated message for thomaskranitsas via Topcoder X```

**Comment by ThomasKranitsas** [comment]
Contest https://www.topcoder.com/challenges/30092403 has been updated - the new changes has been updated for this ticket.<br/><br/>```This is an automated message for thomaskranitsas via Topcoder X```

**Comment by ThomasKranitsas** [comment]
Contest https://www.topcoder.com/challenges/30092403 has been updated - it has been assigned to LieutenantRoger.<br/><br/>```This is an automated message for thomaskranitsas via Topcoder X```

**Comment by ThomasKranitsas** [comment]
Contest https://www.topcoder.com/challenges/30092403 has been updated - it has been assigned to gurmeetb.<br/><br/>```This is an automated message for thomaskranitsas via Topcoder X```

**Comment by gurmeetb** [comment]
@ThomasKranitsas the route cause of this looks like since we are hitting backend onChange of filters, it takes some time for round-trip and if we quickly change the filters, it sends another request to update the filter. 

I think we will see this behavior until we have a waiting spinner which doesn't allows user to change the filters until previous request is complete. I am not sure if we want to take that route.

**Comment by gurmeetb** [comment]
@ThomasKranitsas how do we want to address this?

**Comment by ThomasKranitsas** [comment]
@sushilshinde @Oanh-and-only-Oanh ☝️

**Comment by Oanh-and-only-Oanh** [comment]
@gurmeetb @ThomasKranitsas, go ahead with the spinner solution. Thanks.

Also, what's the average loading time on the first/previous request? Is it a bigger performance issue we want to look at as well?

**Comment by gurmeetb** [comment]
@Oanh-and-only-Oanh thanks for confirming, will try to get spinner implemented in next few hrs and push in PR.

**Comment by Oanh-and-only-Oanh** [comment]
@gurmeetb, we are working on placeholder animation (which would be fixed via https://github.com/topcoder-platform/community-app/issues/2410 ).  Would the placeholder animation work instead of the spinner to keep user from changing the filters until previous request is complete? No need to have both if the placerhold animation is sufficient. @nithyaasworld

**Comment by nithyaasworld** [comment]
Hi @gurmeetb Please check this scenario as well:

1. Go to https://beta-community-app.topcoder.com/challenges?filter%5Btracks%5D%5Bdesign%5D=true&filter%5Btracks%5D%5Bdevelop%5D=true
2. Click on My challenges from the right side
3. Notice that the screen flickers

**Comment by nithyaasworld** [comment]
In the above case, user sends only one request to the backend.

**Comment by gurmeetb** [comment]
@Oanh-and-only-Oanh yes that would work perfect, as I see we plan to show only one animation, is that assumption right seeing the ticket #2410 ? 

I will wait for that to come in, then we can decide on this, I will move this to On Hold till then. 

@nithyaasworld I guess #2410 will take care of this scenario too, since we are doing auto adjust of the screen, Initial screen will have smaller size animation which will expand once we load total no of challenges.
